be/src/storage/rowset/bitmap_index_writer.cpp

```diff
@@ -40,14 +40,64 @@ namespace starrocks {
 
 namespace {
 
+static const size_t roaring_get_size_fastpath_threshold = 1024;
+
+class RoaringWrapper {
+public:
+    explicit RoaringWrapper(rowid_t rid)
+            : _roaring(std::move(Roaring::bitmapOf(1, rid))),
+              _old_size(1),
+              _size_changed(true),
+              _enable_get_size_in_Bytes(false){};
+    Roaring& roaring() { return _roaring; }
+    uint64_t roaring_size() const { return _roaring.getSizeInBytes(false); }
+    uint64_t old_size() const { return _old_size; }
+    void set_old_size(uint64_t size) const { _old_size = size; }
+    bool size_changed() const { return _size_changed; }
+    void set_size_changed() const { _size_changed = true; }
+    void unset_size_changed() const { _size_changed = false; }
+
+    void add_rid(const rowid_t rid, uint64_t& reverted_index_size, vector<RoaringWrapper*>& size_changed_roaring_vec) {
```

Comment:
If you want to change argument's value, please make it a pointer, not a reference.
Is this necessary to get estimated size each time?

Reply:
> Is this necessary to get estimated size each time?

For roaring whose `_element_count` less than `roaring_get_size_fastpath_threshold`, we will not push it into the `size_changed_roaring_vec`. In bitmap index's `size()`, we only update the size of roaring bitmap in `_size_changed_roaring_vec`. 
So here we need to get estimated size each time for roaring whose `_element_count` less than `roaring_get_size_fastpath_threshold`